consensus/consensus-types/src/payload.rs

```diff
@@ -356,64 +429,56 @@ impl OptQuorumStorePayload {
         })
     }
 
-    pub(crate) fn num_txns(&self) -> usize {
-        self.opt_batches.num_txns() + self.proofs.num_txns() + self.inline_batches.num_txns()
-    }
-
-    pub(crate) fn is_empty(&self) -> bool {
-        self.opt_batches.is_empty() && self.proofs.is_empty() && self.inline_batches.is_empty()
-    }
-
-    pub(crate) fn extend(mut self, other: Self) -> Self {
-        let other: OptQuorumStorePayloadV1 = other.into_inner();
-        self.inline_batches.extend(other.inline_batches.0);
-        self.opt_batches.extend(other.opt_batches);
-        self.proofs.extend(other.proofs);
-        self.execution_limits.extend(other.execution_limits);
-        self
-    }
-
-    pub(crate) fn num_bytes(&self) -> usize {
-        self.opt_batches.num_bytes() + self.proofs.num_bytes() + self.inline_batches.num_bytes()
+    pub(crate) fn extend(self, other: Self) -> Self {
+        match (self, other) {
+            (Self::V1(p1), Self::V1(p2)) => Self::V1(p1.extend(p2)),
```

Comment:
Silly question, but do we imagine cases where we're extending V1 and V2 batches together, e.g., during a rollout? Or would they only be exclusive (i.e., only V1 extending other V1s, and V2 extending other V2s?)

Reply:
This extend fn is just for the DAG implementation, where it is possible to have V1 extending V2, etc (we don't use that code path anymore, maybe time for a cleanup?). For our leader based consensus, we only propose one "Payload" and execute it.